pkg/resource/function/hooks.go

```diff
@@ -96,20 +98,21 @@ func (rm *resourceManager) customUpdateFunction(
 	// UpdateFunctionCode because both of them can put the function in a
 	// Pending state.
```

Comment:
I'm seeing a bunch of these in the controller logs:

```
ResourceConflictException: The operation cannot be performed at this time. An update is in progress for resource: arn:aws:lambda:us-west-2:566350749442:function:...
```

You will need to add some sort of check to determine if the function can be updated before calling any updating function, including `rm.updateFunctionCode()` . Check for `Status.State == "Active"` and requeue the resource if not. Otherwise the controller will just constantly try updating a non-updateable resource.